### PR TITLE
Update repository.pp

### PR DESCRIPTION
**Pull request by toerb**
Hey, 

I needed the changes for my centos puppet setup.
 * Puppet version 5.3.3
 * Facter version 2.4.1
 * Centos 7

Please check for debian derivates before merging.

Best
toerb

### CODE REVIEW

File: manifests/repository.pp [hunk 1]
```diff
@@ -11,7 +11,7 @@
   $url = "http://download.bareos.org/bareos/release/${release}/"
 
   $os = $::operatingsystem
-  $osrelease = $::operatingsystemrelease
+  $osrelease = $::os[release][major]
```
Comment (Project0) [inline manifests/repository.pp]: this should fix the error and adds indirect support for old and new puppet release `$osrelease = pick($facts['os']['release']['major'], $::operatingsystemrelease)`

File: manifests/repository.pp [hunk 2]
```diff
@@ -11,7 +11,7 @@
   $url = "http://download.bareos.org/bareos/release/${release}/"
 
   $os = $::operatingsystem
```
Comment (Project0) [inline manifests/repository.pp]: this should fix the error and adds indirect support for old and new puppet release $os = pick($facts['os']['name'], $::operatingsystem)

Comment (toerb) [reply manifests/repository.pp]: works with centos 7

File: manifests/repository.pp (hunk 1 above)
Comment (toerb) [reply manifests/repository.pp]: works with centos 7

File: manifests/repository.pp (hunk 2 above)
Comment (Project0) [reply manifests/repository.pp]: this is fine, but breaks checks and compatibility with puppet 3.x

**Comment by Project0** [comment]
no response

**Pull request closed without merging**